### ISSUE TITLE
Incorrect shadowed label warnings

### ISSUE THREAD

**Issue by withoutboats** [rust-lang/rust#29095]
This may be a byproduct of not having non-lexical lifetimes, but this code generates a bunch of shadowed label warnings that are not accurate (control is not within the block of the allegedly shadowed label at the point the new label is declared):

```
    match 0 {
        0 => {
            'quux: loop {
                break 'quux;
            }
        }
        1 => {
            'quux: loop {
                break 'quux;
            }
        }
        _ => {
            'quux: loop {
                break 'quux;
            }
        }
    }
```

http://is.gd/2yMtC8

As far as I can tell, this warning cannot be turned off (this seems like a bug also).

**Comment by petrochenkov** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/24278
